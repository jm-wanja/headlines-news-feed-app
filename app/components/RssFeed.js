import React from 'react';
import SourcesPanel from './SourcesPanel';
import ArticlesPanel from './ArticlesPanel';
import SortPanel from './SortPanel';
import Login from './LoginPage';
import HeaderPanel from './HeaderPanel';
import { Layout, Drawer, Navigation, Header, Textfield } from 'react-mdl';


export default class RssFeed extends React.Component {
    render() {
        return (
            <div className='rssFeed' >
                {/* <div className='headerPanel'>
                    <HeaderPanel />
                </div>
                <div className='sourcesPanel'>
                    <SourcesPanel />
                </div>
                <div className='articlesPanel'>
                </div>
                <div className='sortPanel'>
                    <SortPanel />
                </div>
                <div className='loginButton'>
                    <GooglePlusLogin />
                </div>*/}
                {/* <Layout fixedHeader fixedDrawer>
                    <HeaderPanel />
                    <SourcesPanel />
                    <ArticlesPanel />
                </Layout>*/}

                <div>
                    <Layout fixedHeader fixedDrawer>
                        <HeaderPanel />

                        <SourcesPanel />
                        <SortPanel />
                        <ArticlesPanel />
                    </Layout>
                </div>

            </div>
        );
    }
}
