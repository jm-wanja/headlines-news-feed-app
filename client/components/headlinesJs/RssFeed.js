import React from 'react';
import SourcesPanel from './SourcesPanel.js';
import ArticlesPanel from './ArticlesPanel.js';
import FiltersPanel from './FiltersPanel.js';
import GooglePlusLogin from './Loginbutton.js';
import HeaderPanel from './HeaderPanel.js'


export default class RssFeed extends React.Component {
    render() {
        return (
            <div className = 'rssFeed'>
                <div className='sourcesPanel'>
                    <SourcesPanel />
                </div>
                <div className='articlesPanel'>
                    <ArticlesPanel />
                </div>
                <div className='filtersPanel'>
                    <FiltersPanel />
                </div>
                <div className='loginButton'>
                    <GooglePlusLogin />
                </div>
                <div classname='headerPanel'>
                    <HeaderPanel />
                </div>
            </div>
        );
    }
}