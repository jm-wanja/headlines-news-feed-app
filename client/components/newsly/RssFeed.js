import React from 'react';
import SourcesPanel from './SourcesPanel.js';
import ArticlesPanel from './ArticlesPanel.js';
import SortPanel from './SortPanel.js';
import GooglePlusLogin from './LoginButton.js';
import HeaderPanel from './HeaderPanel.js'
import { Layout, Drawer, Navigation, Header, Textfield } from "react-mdl";



export default class RssFeed extends React.Component {
    render() {
        return (
            <div className='rssFeed' >
                {/*<div className='headerPanel'>
                    <HeaderPanel />
                </div>
                <div className='sourcesPanel'>
                    <SourcesPanel />
                </div>
                <div className='articlesPanel'>
                </div>
                <div className='sortPanel'>
                    <SortPanel />
                </div>
                <div className='loginButton'>
                    <GooglePlusLogin />
                </div>*/}
                {/*<Layout fixedHeader fixedDrawer>
                    <HeaderPanel />
                    <SourcesPanel />
                    <ArticlesPanel />
                </Layout>*/}

                <div>
                    <Layout fixedHeader fixedDrawer>
                        <HeaderPanel />

                        <SourcesPanel />
                        <SortPanel />
                        <ArticlesPanel />
                    </Layout>
                </div>

            </div>
        );
    }
}