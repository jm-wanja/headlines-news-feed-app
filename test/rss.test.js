import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import RssFeed from '../client/components/headlinesJs/RssFeed.js';

describe('Rss Feed Component', () => {
    it('snapshot of rssFeed', () => {
        const rss = shallow(<RssFeed />);
        const tree = shallowToJSON(rss);
        expect(tree).toMatchSnapshot();
    }),

        it('renders sourcesPanel', () => {
            const sources = shallow(<RssFeed />);
            expect(sources.find('.sourcesPanel')).toBe.defined;
        }),

        it('renders articlesPanel', () => {
            const articles = shallow(<RssFeed />);
            expect(articles.find('.articlesPanel')).toBe.defined;
        }),

        it('renders filterPanel', () => {
            const filters = shallow(<RssFeed />);
            expect(filters.find('.filtersPanel')).toBe.defined;
        }),
        
        it('renders loginButton', () => {
            const login = shallow(<RssFeed />);
            expect(login.find('.loginButton')).toBe.defined;
        }),

        it('renders headerpanel', () => {
            const header = shallow(<RssFeed />);
            expect(header.find('.headerpanel')).toBe.defined;
        })



})