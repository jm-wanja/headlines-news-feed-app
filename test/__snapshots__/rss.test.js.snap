// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rss Feed Component snapshot of rssFeed 1`] = `
<div
  className="rssFeed"
>
  <div
    className="sourcesPanel"
  >
    <SourcesPanel />
  </div>
  <div
    className="articlesPanel"
  >
    <ArticlesPanel />
  </div>
  <div
    className="filtersPanel"
  >
    <FiltersPanel />
  </div>
  <div
    className="loginButton"
  >
    <GooglePlusLogin />
  </div>
  <div
    classname="headerPanel"
  >
    <HeaderPanel />
  </div>
</div>
`;
